R/type-list-of.R

```diff
@@ -68,7 +61,17 @@ new_list_of <- function(x = list(), ptype = logical(), ..., class = character())
     abort("`ptype` must have size 0.")
   }
 
-  new_vctr(x, ..., ptype = ptype, class = c(class, "vctrs_list_of"))
+  new_list_of0(x = x, ptype = ptype, ..., class = class)
+}
+
+new_list_of0 <- function(x, ptype, ..., class = character()) {
+  if (length(class) == 0L) {
```

Comment:
This feels like a micro-optimisation. Does it make a significant difference?

Reply:
Doesn't seem to affect the benchmark in my opening comment so I'll remove